compiler/rustc_middle/src/ty/structural_impls.rs

```diff
@@ -1214,6 +1214,9 @@ impl<'tcx> TypeFoldable<'tcx> for ty::ConstKind<'tcx> {
             ty::ConstKind::Infer(ic) => ty::ConstKind::Infer(ic.try_fold_with(folder)?),
             ty::ConstKind::Param(p) => ty::ConstKind::Param(p.try_fold_with(folder)?),
             ty::ConstKind::Unevaluated(uv) => ty::ConstKind::Unevaluated(uv.try_fold_with(folder)?),
+            // FIXME(julianknodt): need to replace this with walking the nested abstract const?
+            ty::ConstKind::Abstract(_ac) => todo!(),
```

Comment:
well, this branch is just `ty::ConstKind::Abstract(ac.try_fold_with(folder)?)`.

you have to implement `TypeFoldable` for `TyExpr` (or however the new enum is called)